operators/src/source/gdal_source.rs

```diff
@@ -820,771 +691,281 @@ mod tests {
-            .time_tile_iter(
-                origin_split_tileing_strategy,
-                TimeInterval::new_unchecked(0, 1000),
-                query_bbox,
-            )
-            .map(|(time_interval, tile_information)| {
-                gdal_source.load_tile_data(time_interval, tile_information)
-            })
-            .collect();
-        assert_eq!(vres.len(), 2 * 2);
-        let upper_left_pixels: Vec<_> = vres
-            .into_iter()
-            .map(|t| {
-                let raster_tile = t.unwrap();
-
-                raster_tile
-                    .grid_array
-                    .get_at_grid_index([
-                        (tile_size_in_pixels[1] / 2) as isize,
-                        (tile_size_in_pixels[0] / 2) as isize,
-                    ])
-                    .unwrap() // pixel
-            })
-            .collect();
+        assert_eq!(
+            c[1].tile_information().global_tile_position(),
+            [-1, 0].into()
+        );
 
-        let ndvi_center_pixel_values = vec![255, 43, 255, 145];
+        assert_eq!(
+            c[2].tile_information().global_tile_position(),
+            [0, -1].into()
+        );
 
-        assert_eq!(upper_left_pixels, ndvi_center_pixel_values);
+        assert_eq!(
+            c[3].tile_information().global_tile_position(),
+            [0, 0].into()
+        );
     }
 
     #[tokio::test]
-    async fn test_tile_stream_len() {
-        let global_size_in_pixels = [1800, 3600];
-        let tile_size_in_pixels = [600, 600];
-        let dataset_upper_right_coord = (-180.0, 90.0).into();
-        let dataset_x_pixel_size = 0.1;
-        let dataset_y_pixel_size = -0.1;
-        let dataset_geo_transform = GeoTransform::new(
-            dataset_upper_right_coord,
-            dataset_x_pixel_size,
-            dataset_y_pixel_size,
-        );
-
-        let t_1 = TimeInstance::from(NaiveDate::from_ymd(2014, 1, 1).and_hms(0, 0, 0));
-
-        let time_interval_provider = TimeIntervalInformation {
-            start_time: t_1,
-            time_step: TimeStep {
-                granularity: TimeGranularity::Months,
-                step: 1,
-            },
-        };
-
-        let gdal_params = GdalSourceParameters {
-            dataset_id: "modis_ndvi".to_owned(),
-            channel: None,
-        };
-
-        let dataset_information = JsonDatasetInformation {
-            time: time_interval_provider,
-            tile: None,
-            file_name_with_time_placeholder: "MOD13A2_M_NDVI_%%%_START_TIME_%%%.TIFF".into(),
-            time_format: "%Y-%m-%d".into(),
-            base_path: "../modis_ndvi".into(),
-            data_type: RasterDataType::U8,
-            geo_transform: dataset_geo_transform,
-            grid_shape: global_size_in_pixels.into(),
-            spatial_ref: SpatialReference::epsg_4326().into(),
-            measurement: Measurement::Unitless,
-        };
+    async fn test_query_multi_time_slices() {
```

Comment:
how many time slices does this test?

Reply:
two slices. Jan/Feb 2014 each with 4 tiles